source/loader/layers/sanitizer/asan/asan_interceptor.cpp

```diff
@@ -862,10 +884,15 @@ AsanInterceptor::findAllocInfoByAddress(uptr Address) {
         return std::optional<AllocationIterator>{};
     }
     --It;
-    // Make sure we got the right AllocInfo
-    assert(Address >= It->second->AllocBegin &&
-           Address < It->second->AllocBegin + It->second->AllocSize &&
-           "Wrong AllocInfo for the address");
+    // Make sure we got the right AllocInfo, or any other follow-up ops are based on wrong info.
+    bool IsTheRightAllocInfo =
+        Address >= It->second->AllocBegin &&
+        Address < It->second->AllocBegin + It->second->AllocSize;
+    if (!IsTheRightAllocInfo) {
+        getContext()->logger.error(
+            "Failed to find the right AllocInfo for address: {}", Address);
+        die("Address Sanitizer Fatal Error");
+    }
```

Comment:
is it possible that `It == m_AllocationMap.end()`?

Reply:
That is a good question. I will rework the logic here.

Reply:
After thinking, I think that case is not possible here. I have added comments about this.